lib/ngx/re.lua

```diff
@@ -121,24 +130,21 @@ function _M.split(subj, regex, opts, ctx, max, res)
         return res
     end
 
-    if regex == "" then
-        local pos = ctx.pos
-        local last = len
-        if max > 0 then
-            last = math_min(len, pos + max - 1)
-        end
+    local res_idx = 0
+    local sub_idx = ctx.pos
 
-        local res_idx = 1
-        while pos < last do
-            res[res_idx] = sub(subj, pos, pos)
-            res_idx = res_idx + 1
-            pos = pos + 1
-        end
+    local start_regex = find(regex, "^", nil, true) ~= nil
```

Comment:
On `subj = "ab\ncd"` with `/^/` If we don't add the `m` flag, we get:
```
{ "a", "b\ncd" }
```

So we do not match on newlines automatically, but only at position 0, because it is our first character, and the rest doesn't get matched. It seemed to me (after a bit of online research) that Perl's `split` was giving some special care of its own to `/^/`, but I could definitely be wrong.